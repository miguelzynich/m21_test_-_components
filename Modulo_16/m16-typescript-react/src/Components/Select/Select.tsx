import React, { ReactNode } from 'react';
import Flex from '../Flex/Flex';
import 'tachyons/css/tachyons.min.css';
import './Select.css';

enum ActionVariation {
  Primary = 'primary',
}

enum ActionSize {
  Large = 'large',
}

enum ActionFontSize {
  Medium = 'medium',
}

interface SelectProps {
  direction: string;
  variation: ActionVariation;
  size: ActionSize;
  fSize: ActionFontSize;
  id: string;
  labelText: string;
  children: ReactNode;
}

const Select: React.FC<SelectProps> = (props) => {
  const color = `color-action-${props.variation}`;
  const size = `size-action-${props.size}`;
  const fSize = `fSize-action-${props.fSize}`;

  return (
    <Flex direction={props.direction} align="center" justify="center">
      <label className={`f4 ${color} ${fSize} db mb2 mt3 mh3`} htmlFor={props.id}>
        {props.labelText}
      </label>
      <select className={`ba ${color} ${size} ${fSize} f5 fl w-20 mh3`} id={props.id} data-testid="mySelect">
        <option value=''></option>
        {props.children}
      </select>
    </Flex>
  );
}

export default Select;
