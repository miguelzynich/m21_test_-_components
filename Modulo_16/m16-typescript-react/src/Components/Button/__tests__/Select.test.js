import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Select from '../../Select/Select';

test('renders Select component with given props', () => {
  const { getByTestId } = render(
    <Select
      direction="row"
      variation="primary"
      size="large"
      fSize="medium"
      id="mySelect"
      labelText="Select an option"
    >
      <option value="option1">Option 1</option>
      <option value="option2">Option 2</option>
    </Select>
  );

  const selectElement = getByTestId('mySelect');

  expect(selectElement.getAttribute('class')).toBe('ba color-action-primary size-action-large fSize-action-medium f5 fl w-20 mh3');
});

test('renders Select component with missing labelText prop', () => {
  const { queryByTestId, queryByText } = render(
    <Select
      direction="row"
      variation="primary"
      size="large"
      fSize="medium"
      id="mySelect"
    >
      <option value="option1">Option 1</option>
      <option value="option2">Option 2</option>
    </Select>
  );

  expect(queryByTestId('mySelect')).not.toBeNull();
  expect(queryByText('Select an option')).toBeNull();
});

test('Select component matches Snapshot', () => {
  const tree = renderer
    .create(
      <Select
        direction="row"
        variation="primary"
        size="large"
        fSize="medium"
        id="mySelect"
        labelText="Select an option"
      >
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
      </Select>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
