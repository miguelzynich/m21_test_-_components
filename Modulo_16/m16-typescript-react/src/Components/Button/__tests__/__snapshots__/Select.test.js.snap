// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component matches Snapshot 1`] = `
<div
  className="ds-flex direction-action-row align-action-center justify-action-center"
>
  <label
    className="f4 color-action-primary fSize-action-medium db mb2 mt3 mh3"
    htmlFor="mySelect"
  >
    Select an option
  </label>
  <select
    className="ba color-action-primary size-action-large fSize-action-medium f5 fl w-20 mh3"
    data-testid="mySelect"
    id="mySelect"
  >
    <option
      value=""
    />
    <option
      value="option1"
    >
      Option 1
    </option>
    <option
      value="option2"
    >
      Option 2
    </option>
  </select>
</div>
`;
